File: ytrue-admin-server-csharp/Ytrue.Infrastructures/Response/ServerResponseEntity.cs
using Ytrue.Infrastructures.Extensions;

namespace Ytrue.Infrastructures.Response;

/// <summary>
/// 统一返回
/// </summary>
/// <typeparam name="T"></typeparam>
public class ServerResponseEntity<T>
{
    /// <summary>
    /// 自定义的响应码，可以和http响应码一致，也可以不一致
    /// </summary>
    public string Code { get; set; } = null!;

    /// <summary>
    /// 中文消息提示
    /// </summary>
    public string Message { get; set; } = null!;

    /// <summary>
    /// 是否成功
    /// </summary>
    public bool Success { get; set; }

    /// <summary>
    /// 响应的数据
    /// </summary>
    public T? Data { get; set; }


    /// <summary>
    ///  成功
    /// </summary>
    /// <returns></returns>
    public static ServerResponseEntity<T> SuccessResult()
    {
        return new ServerResponseEntity<T>
            { Code = ServerResponseEnum.Ok.GetDescription(), Data = default, Message = "成功", Success = true };
    }

    /// <summary>
    ///  成功
    /// </summary>
    /// <param name="data"></param>
    /// <returns></returns>
    public static ServerResponseEntity<T> SuccessResult(T data)
    {
        return new ServerResponseEntity<T>
            { Code = ServerResponseEnum.Ok.GetDescription(), Data = data, Message = "成功", Success = true };
    }

    /// <summary>
    /// 失败
    /// </summary>
    /// <param name="code"></param>
    /// <param name="message"></param>
    /// <returns></returns>
    public static ServerResponseEntity<T> FailResult(string code, string message)
    {
        return new ServerResponseEntity<T> { Code = code, Data = default, Message = message, Success = false };
    }

    /// <summary>
    /// 失败
    /// </summary>
    /// <param name="message"></param>
    /// <returns></returns>
    public static ServerResponseEntity<T> FailResult(string message)
    {
        return new ServerResponseEntity<T>
        {
            Code = ServerResponseEnum.InternalServerError.GetDescription(), Data = default, Message = message,
            Success = false
        };
    }
}

File: ytrue-admin-server-java/ytrue-infra/ytrue-infra-core/src/main/java/com/ytrue/infra/core/response/ServerResponseEntity.java
package com.ytrue.infra.core.response;

import com.ytrue.infra.core.base.IServerResponseInfo;
import lombok.Data;
import lombok.extern.slf4j.Slf4j;

import java.io.Serializable;
import java.util.Objects;

/**
 * ServerResponseEntity 类用于封装服务器响应的信息，包括状态码、消息和数据。
 *
 * @param <T> 响应数据的类型
 * @author ytrue
 */
@Slf4j
@Data
public class ServerResponseEntity<T> implements Serializable {

    /**
     * 状态码
     */
    private String code;

    /**
     * 信息
     */
    private String message;

    /**
     * 数据
     */
    private T data;

    /**
     * 判断当前响应是否成功。
     *
     * @return 如果状态码为成功，则返回 true；否则返回 false
     */
    public boolean isSuccess() {
        return Objects.equals(ServerResponseInfoEnum.OK.code(), this.code);
    }

    /**
     * 创建一个成功响应，包含数据。
     *
     * @param data 响应数据
     * @param <T>  响应数据的类型
     * @return 成功的 ServerResponseEntity 实例
     */
    public static <T> ServerResponseEntity<T> success(T data) {
        ServerResponseEntity<T> serverResponseEntity = new ServerResponseEntity<>();
        serverResponseEntity.setData(data);
        serverResponseEntity.setCode(ServerResponseInfoEnum.OK.code());
        serverResponseEntity.setMessage(ServerResponseInfoEnum.OK.message());
        return serverResponseEntity;
    }

    /**
     * 创建一个成功响应，不包含数据。
     *
     * @return 成功的 ServerResponseEntity 实例
     */
    public static ServerResponseEntity<Void> success() {
        ServerResponseEntity<Void> serverResponseEntity = new ServerResponseEntity<>();
        serverResponseEntity.setCode(ServerResponseInfoEnum.OK.code());
        serverResponseEntity.setMessage(ServerResponseInfoEnum.OK.message());
        return serverResponseEntity;
    }

    /**
     * 创建一个失败响应，使用默认错误码。
     *
     * @param msg 错误消息
     * @return 失败的 ServerResponseEntity 实例
     */
    public static ServerResponseEntity<Void> fail(String msg) {
        ServerResponseEntity<Void> serverResponseEntity = new ServerResponseEntity<>();
        serverResponseEntity.setMessage(msg);
        serverResponseEntity.setCode(ServerResponseInfoEnum.CUSTOM_FAIL_MESSAGE.code());
        return serverResponseEntity;
    }

    /**
     * 创建一个失败响应，自定义状态码和消息。
     *
     * @param code 自定义状态码
     * @param msg  自定义错误消息
     * @return 失败的 ServerResponseEntity 实例
     */
    public static ServerResponseEntity<Void> fail(String code, String msg) {
        ServerResponseEntity<Void> serverResponseEntity = new ServerResponseEntity<>();
        serverResponseEntity.setCode(code);
        serverResponseEntity.setMessage(msg);
        return serverResponseEntity;
    }

    /**
     * 创建一个失败响应，使用 IServerResponseInfo 中的状态码和消息。
     *
     * @param responseCode 响应信息接口实现
     * @return 失败的 ServerResponseEntity 实例
     */
    public static ServerResponseEntity<Void> fail(IServerResponseInfo responseCode) {
        ServerResponseEntity<Void> serverResponseEntity = new ServerResponseEntity<>();
        serverResponseEntity.setMessage(responseCode.message());
        serverResponseEntity.setCode(responseCode.code());
        return serverResponseEntity;
    }
}
